services/artifactory/artifactory.service.js

```diff
@@ -0,0 +1,211 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { BaseService } from '../index.js'
+import { renderVersionBadge } from '../version.js'
+import { getCachedResource } from '../../core/base-service/resource-cache.js'
+
+const ARTIFACTORY_LATEST_VERSION_ENDPOINT = 'api/search/latestVersion'
+const ONE_HOUR = 1 * 3600 * 1000
+
+const queryParamSchema = Joi.object({
+  // auth related optional properties
+  server: optionalUrl.default(''),
+  token: Joi.string().default(''),
+  username: Joi.string().default(''),
+  password: Joi.string().default(''),
+
+  // api related optional properties
+  repos: Joi.string().default(''),
+  remote: Joi.string().default(''),
+  version: Joi.string().default(''),
+}).required()
+
+const documentation = `
+  <p>
+    Query for the last released version of an artifact using the <b>Latest Version</b> endpoint.
+    <br />
+    <br />
+    This service will cache results for <b>1 hour</b> so as not to overload a given Artifactory instance with potentially long running query operations.
+    <br />
+    <br />
+    Authentication is done either through basic auth or using a bearer token. While the options exist to pass along credentials as query params, users
+    are strongly encouraged to instead set the relevant auth related environment variables server side which this service will pick up and read at runtime.
+    <br />
+    <br />
+    The following environment variables are supported:
+    <ul>
+      <li>ARTIFACTORY_URL</li>
+      <li>ARTIFACTORY_TOKEN</li>
+      <li>ARTIFACTORY_USERNAME</li>
+      <li>ARTIFACTORY_PASSWORD</li>
+    </ul>
+    If using a <b>Token</b> for authentication you do not need to set the username and password env-vars.
+    <br />
+    <br />
+    For more technical information on the available properties the <b>Latest Version</b> endpoint uses, and their potential drawbacks and limitations, please consult the following <a target="_blank" href="https://jfrog.com/help/r/jfrog-rest-apis/artifact-latest-version-search-based-on-layout">developer documentation here</a>.
+  </p>
+  `
+
+export default class Artifactory extends BaseService {
+  static category = 'version'
+  static route = {
+    base: 'artifactory/v',
+    pattern: ':group/:artifact?',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Artifactory Latest Version',
+      documentation,
+      namedParams: {
+        group: 'org.jfrog.artifactory.client',
+        artifact: 'artifactory-java-client-services',
+      },
+      queryParams: {
+        repos: 'libs-release-local',
+        remote: '1',
+        version: '1.0-SNAPSHOT',
+        server: 'https://my.artifactory.server/artifactory',
+        token: 'OTEfj876SDF2345JKIB/H&+FFGH',
+        username: 'jerry',
+        password: 'seinfeld',
+      },
+      staticPreview: Artifactory.render('2.1.0'),
```

Comment:
No matter what I did I couldn't get any of this to show up in the local "integ test server". I assume I'm doing something wrong but filled out all the relevant links/options/docs/etc anyway if only for completeness sake.